apio-consumer/src/main/kotlin/com/liferay/apio/consumer/ApioConsumer.kt

```diff
@@ -38,22 +36,29 @@ class ApioConsumer @JvmOverloads constructor(authenticator: ApioAuthenticator? =
 		RequestAuthorization.authenticator = authenticator
 	}
 
-	fun fetch(url: HttpUrl, authenticator: ApioAuthenticator? = null, fields: Map<String, List<String>> = emptyMap(),
-		embedded: List<String> = emptyList(), onSuccess: (Thing) -> Unit,
-		onError: (Exception) -> Unit = emptyOnError()) {
+	@JvmOverloads
+	fun fetch(url: HttpUrl, authenticator: ApioAuthenticator? = RequestAuthorization.authenticator,
+		fields: Map<String, List<String>> = emptyMap(), embedded: List<String> = emptyList(), callback: ApioCallback) {
+
+		fetch(url, authenticator, fields, embedded, callback::onComplete)
+	}
+
+	@JvmSynthetic
+	fun fetch(url: HttpUrl, authenticator: ApioAuthenticator? = RequestAuthorization.authenticator,
+		fields: Map<String, List<String>> = emptyMap(), embedded: List<String> = emptyList(),
+		onSuccess: (Thing) -> Unit, onError: (Exception) -> Unit = emptyOnError()) {
 
 		fetch(url, authenticator, fields, embedded) {
 			it.fold(onSuccess, onError)
 		}
 	}
 
-	@JvmOverloads
-	fun fetch(url: HttpUrl, authenticator: ApioAuthenticator? = null, fields: Map<String, List<String>> = emptyMap(),
-		embedded: List<String> = emptyList(), onComplete: (Result<Thing, Exception>) -> Unit = emptyOnComplete()) {
+	@JvmSynthetic
+	fun fetch(url: HttpUrl, authenticator: ApioAuthenticator? = RequestAuthorization.authenticator,
```

Comment:
What do you think @nhpatt @victorg1991? I created an ApioCallback interface to easily the life of Java developers, it wasn't good implement `Function1`to handle callbacks, so the other overloads are only for Kotlin since it uses blocks.

Reply:
I like it :)